src/sentry/scim/endpoints/members.py

```diff
@@ -251,6 +252,85 @@ def delete(self, request: Request, organization, member) -> Response:
         self._delete_member(request, organization, member)
         return Response(status=204)
 
+    @extend_schema(
+        operation_id="Update an Organization Member's Attributes",
+        parameters=[GLOBAL_PARAMS.ORG_SLUG, SCIM_PARAMS.MEMBER_ID],
+        request=inline_serializer(
+            "SCIMMemberProvision", fields={"sentryOrgRole": serializers.CharField()}
+        ),
+        responses={
+            201: OrganizationMemberSCIMSerializer,
+            401: RESPONSE_UNAUTHORIZED,
+            403: RESPONSE_FORBIDDEN,
+            404: RESPONSE_NOTFOUND,
+        },
+        examples=[  # TODO: see if this can go on serializer object instead
+            OpenApiExample(
+                "Update a user",
+                response_only=True,
+                value={
+                    "schemas": ["urn:ietf:params:scim:schemas:core:2.0:User"],
+                    "id": "242",
+                    "userName": "test.user@okta.local",
+                    "emails": [{"primary": True, "value": "test.user@okta.local", "type": "work"}],
+                    "active": True,
+                    "name": {"familyName": "N/A", "givenName": "N/A"},
+                    "meta": {"resourceType": "User"},
+                },
+                status_codes=["201"],
+            ),
+        ],
+    )
+    def put(self, request: Request, organization, member):
+        """
+        Update an organization member
+
+        Currently only updates organization role
+        """
+        if features.has("organizations:scim-orgmember-roles", organization, actor=None):
+
+            if "sentryOrgRole" in request.data and request.data["sentryOrgRole"]:
+                # Only do an update if the org role is changing
+                if (
+                    member.flags["idp:role-restricted"]
+                    and member.role.lower() == request.data["sentryOrgRole"].lower()
+                ):
+                    context = serialize(
+                        member, serializer=_scim_member_serializer_with_expansion(organization)
+                    )
+                    return Response(context, status=200)
+                requested_role = request.data["sentryOrgRole"].lower()
+                idp_role_restricted = True
+            else:
+                # Only reset role if current role isn't the default
+                if (
+                    member.role != organization.default_role
+                    and not member.flags["idp:role-restricted"]
```

Comment:
If the role is restricted, shouldn't this api still modify the role? iiuc this api is what the IDP would use to provision roles right?

Reply:
this block is covering the case where the member isn't idp_role_restricted and you set the role in the IdP to blank --> nothing changes
https://github.com/getsentry/sentry/blob/d58431b5e553be7b6acd94be2e32fab1b52a7e93/tests/sentry/api/endpoints/test_scim_user_details.py#L148-L149

Reply:
also, here the member is not role-restricted. i'll add a comment to clarify though